spec/models/king_spec.rb

```diff
@@ -25,4 +25,46 @@
       expect(king.valid_move?(Coordinates.new(0, 4))).to eq false
     end
   end
+
+  describe '#can_castle' do
+    let(:game) { create(:game) }
+    let(:king) { game.chess_pieces.find_by(position_x: 5, position_y: 8) }
+    let(:rook) { game.chess_pieces.find_by(position_x: 1, position_y: 8) }
+
+    it 'returns false if a rook has previously moved' do
+      remove_everything_but_rook_and_king!('black')
+      rook.move_to!(Coordinates.new(1, 4))
+
+      expect(king.can_castle?(rook)).to eq false
+    end
+
+    it 'returns false if the king has previously moved' do
+      remove_everything_but_rook_and_king!('black')
+      king.move_to!(Coordinates.new(5, 7))
+
+      expect(king.can_castle?(rook)).to eq false
+    end
+
+    it 'returns false if, when trying to castle queenside, there is an obstruction' do
+      expect(king.can_castle?(rook)).to eq false
+    end
+
+    it 'returns false if, when trying to castle kingside, there is an obstruction' do
```

Comment:
Line is too long. [85/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.